src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -330,6 +330,30 @@ public ShardIterator onlyNodeActiveInitializingShardsIt(String nodeId) {
         return new PlainShardIterator(shardId, ordered);
     }
 
+    public ShardIterator onlyNodeSelectorActiveInitializingShardsIt(String nodeSelector,DiscoveryNodes discoveryNodes) {
+        ArrayList<ShardRouting> ordered = new ArrayList<>(activeShards.size() + allInitializingShards.size());
+        List<String> selectedNodes = Arrays.asList(discoveryNodes.resolveNodesIds(nodeSelector));
```

Comment:
can this be a `Set<String>` instead I think we should not do linear access `#contains` calls